src/scripts/breeding/Egg.ts

```diff
@@ -8,6 +8,8 @@ class Egg implements Saveable {
     shinySteps: number;
     pokemon: string;
     type: GameConstants.EggType;
```

Comment:
Should this regular type still be used?

Reply:
Unless we stop showing fossils as their fossil image, we need this type to differentiate between normal breeding and EggType.Fossil

This enum is also used when choosing and setting up pokemon from shop bought eggs.

It seems like this enum is two things at once: shopEggType and eggDisplayType. 
We should probably separate it into these two things and just pass the displayType into the constructor (Pokemon or Fossil right now, but potentially something else in the future), and we can rename the property to reflect this.